arch/x86/compare258_avx.c

```diff
@@ -16,29 +16,29 @@
 #endif
 
 /* UNALIGNED_OK, AVX2 intrinsic comparison */
-static inline int32_t compare256_unaligned_avx2_static(const unsigned char *src0, const unsigned char *src1) {
-    int32_t len = 0;
+static inline uint32_t compare256_unaligned_avx2_static(const unsigned char *src0, const unsigned char *src1) {
+    uint32_t len = 0;
 
     do {
         __m256i ymm_src0, ymm_src1, ymm_cmp;
         ymm_src0 = _mm256_loadu_si256((__m256i*)src0);
         ymm_src1 = _mm256_loadu_si256((__m256i*)src1);
         ymm_cmp = _mm256_cmpeq_epi8(ymm_src0, ymm_src1); /* non-identical bytes = 00, identical bytes = FF */
-        int mask = _mm256_movemask_epi8(ymm_cmp);
-        if ((unsigned int)mask != 0xFFFFFFFF) {
-            int match_byte = __builtin_ctz(~mask); /* Invert bits so identical = 0 */
-            return (int32_t)(len + match_byte);
+        unsigned mask = (unsigned)_mm256_movemask_epi8(ymm_cmp);
+        if (mask != 0xFFFFFFFF) {
+            uint32_t match_byte = (uint32_t)__builtin_ctz(~mask); /* Invert bits so identical = 0 */
+            return (uint32_t)(len + match_byte);
         }
 
         src0 += 32, src1 += 32, len += 32;
 
         ymm_src0 = _mm256_loadu_si256((__m256i*)src0);
         ymm_src1 = _mm256_loadu_si256((__m256i*)src1);
         ymm_cmp = _mm256_cmpeq_epi8(ymm_src0, ymm_src1);
-        mask = _mm256_movemask_epi8(ymm_cmp);
-        if ((unsigned int)mask != 0xFFFFFFFF) {
-            int match_byte = __builtin_ctz(~mask);
-            return (int32_t)(len + match_byte);
+        mask = (unsigned)_mm256_movemask_epi8(ymm_cmp);
+        if (mask != 0xFFFFFFFF) {
+            uint32_t match_byte = (uint32_t)__builtin_ctz(~mask);
+            return (uint32_t)(len + match_byte);
```

Comment:
```len``` and ```match_byte``` are both uint32_t, so the cast is superfluous.

Reply:
Fixed